### PR TITLE
Adds an additional zoom visibility property to markers to show/hide b…

### PR DESCRIPTION
**Pull request by digitaldan**
…ased on current zoom level.

Signed-off-by: digitaldan <dan@digitaldan.com>

### CODE REVIEW

**Comment by digitaldan** [comment]
zoomVisibility

**Comment by digitaldan** [comment]
Hi @ghys was playing around with a way to be able to zoom into a map and display more markers based on the zoom level.  Not sure if this is the best way to handle that. or if i event did it right :-)

**Comment by ghys** [comment]
Thanks @digitaldan, that's a nice feature! My only (small) concern is that users would need to figure out by themselves what the zoom level is, since it's neither shown anywhere nor retrieved easily, so it's not an option for everyone. Therefore I'd like to suggest making it an "advanced" option, meaning it will only be shown after checking a "show advanced options" box. Doing so is easy, the widgets have the same config parameter description format as other parts in openHAB so you just have to add `advanced: true` in your description, the checkbox will appear automatically.

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-marker.vue [hunk 1]
```diff
@@ -136,7 +136,13 @@ export default {
...
           context: 'color',
           description: 'Color of the tooltip'
-        }
+        },
+        {
+          name: 'zoomVisibility',
+          label: 'Zoom Visibility',
+          type: 'NUMBER',
+          description: 'Visible only when zoomed to at least this level (empty by default)'
+        },
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-marker.vue]: Please remove the trailing comma here, and add `advanced: true,` above (3rd line or so) as discussed above.

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue [hunk 2]
```diff
@@ -170,6 +170,14 @@ export default {
       zoomControl:false
       } : {})
     },
+    markers(){
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]: Please add spaces before and after the braces.

**Comment by ghys** [comment]
@digitaldan actually there quite a few linting errors in this and your previous PR (I overlooked it last time) 😅

image

I have a laxer approach than most of the other openHAB projects when it comes to enforcing them, in that they would not fail the build, but still it would be great if could sort them out while you're at it 😄.

If you're using VS Code as your IDE, searching for ESLint extension and installing `dbaeumer.vscode-eslint` should do the trick. When developing Vue files you'll want to install Vetur (`octref.vetur`) as well.

Then it should only be a matter of looking for the little lightbulb icons when hovering over code or in the problems window, and selecting 'Fix all auto-fixable problems':

image

Let me know how this works for you so we can eventually add this to the CONTRIBUTING file.

**Comment by digitaldan** [comment]
>My only (small) concern is that users would need to figure out by themselves what the zoom level is

I was thinking the same thing.  I had played with displaying the zoom level when editing, but i think making it an advanced property makes sense. I may still play with displaying something.

>I have a laxer approach than most of the other openHAB projects when it comes to enforcing them

I would not mind make making this stricter actually. There is a lot of code, and it really helps to have consistency.  I don't mind making sure the code is properly linted before checking in.. 

> extension pane and installing dbaeumer.vscode-eslint

Nice, that worked perfectly.  I will start paying attention to the "problems" tab now :-)

**Comment by digitaldan** [comment]
zoomVisibility

I added a zoom display when editing, thoughts?

**Comment by ghys** [comment]
> I added a zoom display when editing, thoughts?

I like it! ;)

> I would not mind make making this stricter actually. There is a lot of code, and it really helps to have consistency. I don't mind making sure the code is properly linted before checking in..

Yes, that could be valuable, you could go further and add the `vue/recommended` rules for [stricter rules](https://vuejs.github.io/eslint-plugin-vue/rules/) (right now there's only `vue/essential` and `@vue/standard` in .eslintrc), also why not trying out [Prettier](https://prettier.io), which is widely praised, as well and checking a default VS Code settings file to fix problems on save.
Some details here: https://medium.com/@gogl.alex/how-to-properly-set-up-eslint-with-prettier-for-vue-or-nuxt-in-vscode-e42532099a9c.
For now I was happy with only having to fix basic problems highlighted in the editor but if there are more contributions over time we should probably have a plan to keep the code clean. I just don't want to deal with the backlog right now 😄

**Comment by digitaldan** [comment]
FYI, there was an issue with markers flickering when zooming, my last push fixes this, but may seem counterintuitive at first glance

**Comment by digitaldan** [comment]
Ignore my last comment and commit, there was something else goofy going on with my browser.

**Pull request merged**